### DISCUSSION TITLE
[Bug]: TypeError [ERR_INVALID_ARG_TYPE]

### DISCUSSION THREAD

**Discussion by Zapzatron** [danny-avila/LibreChat#3016] [Troubleshooting]
### What happened?

Got an error `TypeError [ERR_INVALID_ARG_TYPE]` when embedding LibreChat into Hugging Face Spaces.
[My space at Hugging Face](https://huggingface.co/spaces/Superchik/LibreChat)

### Steps to Reproduce

1. Embed a LibreChat space in your space
2. Wait for a build error
3. Get an error

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
===== Application Startup at 2024-06-09 19:03:13 =====


> LibreChat@0.7.2 backend
> cross-env NODE_ENV=production node api/server/index.js

node:buffer:319
  throw new ERR_INVALID_ARG_TYPE(
  ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (node:buffer:319:9)
    at Object.<anonymous> (/app/api/server/utils/crypto.js:4:20)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/app/api/server/utils/index.js:7:21)
    at Module._compile (node:internal/modules/cjs/loader:1358:14) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v20.14.0
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
Looks like you didn't provide all required `secrets`

**Comment by fuegovic** [comment]
Also, you should try cloning this template instead of our deployed instance, then customize it.

https://huggingface.co/spaces/LibreChat/template

**Comment by Zapzatron** [comment]
![image](https://github.com/danny-avila/LibreChat/assets/100136305/b5942f8a-0a8d-40f6-a7e6-724f6254e891)
Do I need to add these keys to the `secrets`?

**Comment by Zapzatron** [comment]
I cloned from [LibreChat/LibreChat](https://huggingface.co/spaces/LibreChat/LibreChat)

**Reply by fuegovic** [reply]
all secrets need to be filled, that's why you should start with the template (like I commented above) instead of cloning our demo. The template config is way simpler and once you get it to run you can add more custom options to it

**Reply by fuegovic** [reply]
if you haven't seen this yet:
https://www.librechat.ai/docs/remote/huggingface

**Comment by Zapzatron** [comment]
![image](https://github.com/danny-avila/LibreChat/assets/100136305/03758fc1-d75b-4153-9ed0-b1dff19f19d6)
I also tried using your space but it doesn't work.
(Can't log in with e-mail address and password)

**Reply by fuegovic** [reply]
did you create an account first?
![image](https://github.com/danny-avila/LibreChat/assets/32828263/54e81698-d77f-447c-be60-12c69add367f)

I just tested and was able to login with a new account I created...

**Reply by Zapzatron** [reply]
Yes, I created an account first

**Reply by Zapzatron** [reply]
I tried to sign up with google first but the button doesn't work. And then it worked to register via password but can't login

**Reply by fuegovic** [reply]
try login with google from this URL:
https://librechat-librechat.hf.space/login

(the social login won't work from the embedded space page, this url gives you direct acces)

**Reply by Zapzatron** [reply]
I understand correctly that you can't add your own endpoint?
I have my own API and wanted to add its endpoint.

**Reply by fuegovic** [reply]
You can, if you deploy your own, with the librechat.yaml config file

**Comment by Zapzatron** [comment]
```
> LibreChat@0.7.2 backend
> cross-env NODE_ENV=production node api/server/index.js

2024-06-09 22:17:02 info: [Optional] Redis not initialized. Note: Redis support is experimental.
2024-06-09 22:17:03 info: Starting Agenda...
2024-06-09 22:17:03 error: KeyvMongo connection error: URI must include hostname, domain name, and tld
2024-06-09 22:17:03 error: There was an uncaught error: URI must include hostname, domain name, and tld
```
I added the link to MONGO_URI

**Reply by Zapzatron** [reply]
![image](https://github.com/danny-avila/LibreChat/assets/100136305/48fbe4ae-b8ed-4c64-a587-b956a9b7123e)

**Reply by fuegovic** [reply]
Looks like your mongo uri is invalid, make sure you follow: https://www.librechat.ai/docs/configuration/mongodb/mongodb_atlas

**Reply by Zapzatron** [reply]
The problem was in the password to access the database. I changed it to be without special characters.

**Comment by Zapzatron** [comment]
Did I understand correctly that a user cannot add their baseURL and apiKey?

**Reply by fuegovic** [reply]
You can set both the baseURL and apiKey to "user_provided" in your librechat.yaml config file, however the model list needs to be pre-configured.

**Comment by husniadil** [comment]
I have successfully set it up and got it running on `huggingface`, but I noticed a lack of the gpt-4o option. This might be because the docker image has not been rebuilt with the latest changes from the main branch. Or maybe the "template" itself? CMIIW.

**Reply by fuegovic** [reply]
The models are listed in your variables, you can add it yourself:

![image](https://github.com/danny-avila/LibreChat/assets/32828263/c8ecfa29-5738-4c54-9f1d-57a6821a1c4c)
